climada/util/plot.py

```diff
@@ -290,6 +293,79 @@ def geo_im_from_array(array_sub, coord, var_name, title,
+                            shapes=True, axes=None, **kwargs):
+    """Map plots for categorical data defined in array(s) over input
+    coordinates. The categories must be a finite set of unique
+    values as can be identified by np.unique() (int, float, strings, ...).
+    
+    The categories are shared among all subplots, i.e. are obtained from 
+    np.unique(array_sub). 
+    Example:
+        array_sub = [[1, 2, 1.0, 2], [1, 2, 'a', 'a']] 
+        -> categories mapping is [[0, 2, 1, 2], [0, 2, 3, 3]]
+    
+    
+    Parameters:
+        array_sub (np.array(1d or 2d) or list(np.array)): Each array (in a row
+            or in  the list) are categorical values at each point
+            in corresponding geo_coord that are ploted in one subplot.
+        coord (2d np.array): (lat, lon) for each point in a row.
+            The same grid is used for all subplots.
+        var_name (str or list(str)): label to be shown in the colorbar. If one
+            provided, the same is used for all subplots. Otherwise provide as
+            many as subplots in array_sub.
+        cat_name (list(str)), optional: List of category names in same order
+            as the ordered categories obtain from np.unique(array_sub). All
+            subplots are assumed to have the same categories. 
+            Default is numbers (1,2,...)
+        title (str or list(str)): subplot title. If one provided, the same is
+            used for all subplots. Otherwise provide as many as subplots in
+            array_sub.
+        proj (ccrs): coordinate reference system used in coordinates
+        smooth (bool, optional): smooth plot to RESOLUTIONxRESOLUTION. Default:
+            True.
+        kwargs (optional): arguments for pcolormesh matplotlib function.
+
+    Returns:
+        cartopy.mpl.geoaxes.GeoAxesSubplot
+    """
+    
+    #default cmap
+    cmap_name = 'Accent'
+    if 'cmap' in kwargs:
+        cmap_name = kwargs['cmap']
+        del kwargs['cmap']
+        
+    # define the colormap
+    cmap = mpl.cm.get_cmap(cmap_name)       
+    # convert categories to numeric array [0, 1, ...]
+    array_sub_cat = u_vrpr.val_to_cat(array_sub) 
+    bounds = np.arange(-0.5, np.unique(array_sub_cat)[-1] + 1, 1) #np.unique sorts values
+    norm = mpl.colors.BoundaryNorm(bounds, cmap.N)
+    
+    #create the axes
+    kwargs['cmap'] = cmap
+    kwargs['norm'] = norm
+    axes = geo_scatter_from_array(array_sub_cat, geo_coord, var_name, title,
+                            pop_name=pop_name, buffer=buffer,
+                            extend=extend, proj=proj,
+                            shapes=shapes, axes=axes, **kwargs)
+    
+    #add colorbar labels
+    if not isinstance(axes, np.ndarray):
+        axes = [axes]
+    for ax in axes:
+        cbar = ax.collections[-1].colorbar
+        cbar.set_ticks(np.unique(array_sub_cat))
+        if cat_name:
+            cbar.set_ticklabels(cat_name)
```

Comment:
```suggestion
        cbar = ax.collections[-1].colorbar
        cbar.set_ticks(np.arange(array_sub_n))
        cbar.set_ticklabels(cat_name if cat_name else array_sub_unique)
```

Reply:
If the parameter `cat_name` is a dict, you would do: 
```
    cbar.set_ticklabels([cat_name[val] for val in array_sub_unique])
```

Reply:
Yes. Now I tend to think that maybe not specifying the labels is better?

Reply:
Maybe you are right. But if you want to write some kind of tutorial for how to do this, you can just as well remove the whole function and just add a tutorial on how to use `geo_scatter_from_array` or any other plotting function with discrete value ranges.

Reply:
Hmmm I think this is a bit extreme. I mostly wrote this function because I want to use in the `uncertainty` module and tought I might not be the only one. Hence, I think this function is useful.

Reply:
From my perspective, this function is a wrapper around `geo_scatter_from_array` that applies a discrete scaling. However, I don't see, why we shouldn't implement it in a way so that it can be applied to an arbitrary plotting function that uses colormaps. Exlicitly repeating all the parameters of `geo_scatter_from_array` in the function header together with all the docstrings is quite some overhead.

Reply:
@zeliest : what would be your preference? The possibility to specify `cat_name` or not?

Reply:
Aaa now I understand your idea. But the potential issue I see is that for instance lets say you want to wrap around `geo_im_from_array`. I think this will not work well, as the latter makes an interpolation. 

This is why I did not go the route of a general wrapper function...

Reply:
I do think it makes sense to be able to specify it. I think that there are a lot of applications where the categories wouldn't directly be the intensity.

Reply:
OK, then we can keep it. Note that intensity will almost never be a good candidate for a categoric plot, as it is usually a continuous variable in CLIMADA.